dot/parachain/overseer/mockable_overseer.go

```diff
@@ -56,30 +57,40 @@ func (m *MockableOverseer) Start() error {
 
 func (m *MockableOverseer) Stop() {
 	m.cancel()
+	close(m.overseerToSubsystem)
 }
 
+// ReceiveMessage method is to receive overseer messages in a subsystem which we are testing
 func (m *MockableOverseer) ReceiveMessage(msg any) {
 	m.overseerToSubsystem <- msg
 }
 
-func (m *MockableOverseer) ExpectMessageWithAction(msg any, fn func(msg any)) {
-	m.expectedMessagesWithAction[msg] = fn
+// ExpectActions method is to set expected actions for overseer messages we receive from the subsystem.
+// actions are expected in the order they are set.
+// all the functions in the arguments should return false if the message is unexpected.
+func (m *MockableOverseer) ExpectActions(fns ...func(msg any) bool) {
+	m.actions = append(m.actions, fns...)
 }
 
 func (m *MockableOverseer) processMessages() {
+	actionIndex := 0
 	for {
 		select {
 		case msg := <-m.SubsystemsToOverseer:
 			if msg == nil {
 				continue
 			}
-			action, ok := m.expectedMessagesWithAction[msg]
-			if !ok {
-				m.t.Errorf("unexpected message: %v", msg)
-				continue
-			}
```

Comment:
why not t.Error in case of an unexpected message?

Reply:
If you look at the updated code, we are doing it.